src/movies.js

```diff
@@ -1,16 +1,93 @@
 // Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+
+function getAllDirectors(moviesArr) {
+  let directorsList = moviesArr.map((movie) => {
+    return movie.director;
+  });
+  return directorsList;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(arr) {
+  let dramaMovies = arr.filter((movie) => {
+    return movie.genre.includes("Drama");
+  });
+  let spielbergDramas = dramaMovies.filter((movie) => {
+    return movie.director.includes("Steven Spielberg");
+  });
+  return spielbergDramas.length;
+}
 
```

Comment:
Good work Rich!

Here's an alternative way to write function `howManyMovies`

`function howManyMovies(someMovies) {
 
 return someMovies.filter(
    eachMovie =>
      eachMovie.director === 'Steven Spielberg' &&
      eachMovie.genre.includes('Drama')
  ).length;
}`